### PR TITLE
qe: different approach to prevent napi code from being compiled with query-engine binary

### PR DESCRIPTION
**Pull request by miguelff**
This patch reverts #4035 to implement a different approach to prevent the Node-API code from ending up in the query-engine binary, thus causing linker errors.

To prevent conflicts in https://github.com/prisma/prisma-engines/pull/4031 the cosmetic refactoring of crate names and features will be performed in future PR.

### CODE REVIEW

File: query-engine/connectors/sql-query-connector/src/database/nodejs.rs [hunk 1]
```diff
@@ -2,15 +2,17 @@ use async_trait::async_trait;
 use quaint::{
     connector::IsolationLevel,
     pooled::{PooledConnection, Quaint},
-    prelude::{Query, Queryable, TransactionCapable},
+    prelude::{BoxedQueryable, Query, Queryable, TransactionCapable},
     Value,
 };
 
+use crate::SqlError;
```
Comment (miguelff) [inline query-engine/connectors/sql-query-connector/src/database/nodejs.rs:9]: TODO: Consolidate imports

File: query-engine/connectors/sql-query-connector/src/database/mysql.rs [hunk 2]
```diff
@@ -63,12 +46,11 @@ impl FromSource for Mysql {
         if source.provider == "@prisma/mysql" {
             #[cfg(feature = "nodejs-drivers")]
             {
-                let queryable = nodejs_drivers::installed_driver().unwrap().clone();
+                let pool = super::js::registered_driver();
```
Comment (miguelff) [inline query-engine/connectors/sql-query-connector/src/database/mysql.rs:49]: rename this

File: query-engine/query-engine-node-api/src/engine.rs [hunk 3]
```diff
@@ -147,15 +148,19 @@ impl QueryEngine {
...
     ) -> napi::Result<Self> {
         let log_callback = LogCallback::new(napi_env, callback)?;
         log_callback.unref(&napi_env)?;
 
         // Initialize the global NODEJS_QUERYABLE from fn_ctx.
         // This implies that there can only be one QueryEngine instance per process.
-        if let Some(ctx) = fn_ctx {
-            nodejs_drivers::install_driver(ctx);
+        if let Some(driver) = maybe_driver {
+            let queryable = Queryable::from(driver);
```
Comment (miguelff) [inline query-engine/query-engine-node-api/src/engine.rs:159]: replace import with explicit namespace

**Review by jkomyno — APPROVED** [review]

**Review by Weakky — APPROVED** [review]
LGTM 👍

**Pull request merged**